### ISSUE TITLE
Where have all my notes gone?

### ISSUE THREAD

**Issue by simonkeng** [Laverna/laverna#440]
I have an encryption password and sync my notes to dropbox, but this morning I opened my web browser and went to laverna and the site didnt ask for my encryption and all of my notes are gone. What do I do? I do I retrieve my notes and log in with my encryption code? Does anyone know why this happened? 

Thanks

**Comment by wwebfor** [comment]
Can you check if encryption is enabled by opening settings page? If it isn't, try to import settings from the other instance.

**Comment by simonkeng** [comment]
Encryption is not enabled, if I enable it and enter in the original encryption password I created, would all of my notes appear again? I tried to "import setting" and "import everything", using the .json files that Laverna saved to my dropbox account but when I do this nothing happens. I click save and go back to my notes and there is still nothing there.

**Comment by wwebfor** [comment]
Have you tried to restart the page?

**Comment by simonkeng** [comment]
Yes I have. I have only been using Laverna through a web browser. Originally I created an encryption and began making notes for maybe two days. Then I logged onto my browser and went to laverna, but I didnt ask me for my encryption and instead thought I was starting as a new user. I am just trying to figure out how to access the original notes I made.

**Comment by simonkeng** [comment]
I just went on and again and clicked to use the encryption, I entered my original encryption code in hopes that it would realize who I am from the code so I could get my notes, but it just opened me a new laverna page that still didn't have my notes.

**Comment by wwebfor** [comment]
What browser do you use?

**Comment by simonkeng** [comment]
Google Chrome on OS X

**Comment by simonkeng** [comment]
I have tried multiple times to upload the .json files that Laverna saved to my dropbox but everytime I try to use the "import settings" or "import everything" buttons, nothing happens afterwards. I have tried a mixture of saving after, reloading the page, trying some troubleshooting of my own but nothing is working. I tested out Laverna for taking notes on some upcoming exams, after spending a whole day using it, now I am desperate to retrieve those notes that had details on topics I need to study. Any assistance to retrieve my notes would be very appreciated! Thank you.

**Comment by claell** [comment]
So since you set up Dropbox sync hopefully your notes are not lost. I would suggest to create a copy of the corresponding folder on Dropbox to make sure they are not overwritten, also check the dates and size to verify that the notes are still there.

After that I would go in settings of laverna and make sure that Dropbox sync is enabled.

You could also try to setup the sync on a different browser, since maybe the configuration of your main browser is broken.

**Comment by wwebfor** [comment]
What version of the app are you using?

**Comment by simonkeng** [comment]
Laverna Beta v0.7.1 on my OS X, but I was using the web browser (Chrome) originally when I made several notes and then opened my browser on a different day and couldn't find them.

**Comment by wwebfor** [comment]
Maybe you used a different version on the other browser? Try to Laverna folder on Dropbox and open laverna.cc/nightly.

**Comment by hasues** [comment]
Did you happen to use a "Profile"?  If you made a profile, you would have to make sure to select the profile you were using as the default one will not contain the notes you were saving?

**Comment by dlip** [comment]
I'm having a similar issue, I'm using linux desktop 0.7.2-RC4 now but I've had the this problem with earlier versions. Steps:

1. Delete old apps/laverna folder on dropbox and .config/laverna folder
2. Run laverna and connect to dropbox
3. Create note a, note b, then sync and close laverna
4. Delete .config/laverna folder
5. Run laverna and connect to dropbox again
6. Notes are empty

I noticed that in dropbox there was still 2 note files in the notes-db/notes folder and when i create a new note it is added to this folder. It is like it just ignores the existing notes. Maybe the password salt needs to be the same between installs?

**Comment by wweich** [comment]
If you encrypt your notes and use a different salt, the notes cannot be decrypted.

**Comment by dlip** [comment]
So is the general approach to export your settings file and keep it safe somewhere outside of dropbox?

**Comment by Boggin** [comment]
This makes it too easy to lose work. A user would assume they needed only the location of their sync'd files and their encryption password. Requiring a export and import of the salt isn't intuitive. I understand the salt is necessary for key stretching but it's still not clear how you log in from a different machine.

(1) Document how to open your work on a second machine when your notes are encrypted.
(2) Add a warning in the Encryption settings screen, e.g. "You must remember your password and you must either remember the salt or export your settings."
(3) Make a workflow around enabling and updating encryption that forces the user to consider how they are going to record their password and their salt.

And yes, I lost work when I cleared local storage on my machine.

**Comment by Poorchop** [comment]
>(2) Add a warning in the Encryption settings screen, e.g. "You must remember your password and you must either remember the salt or export your settings."

This should go without saving. Why would you need someone to explicitly remind you to remember your password? Do you often create passwords without any intention of memorizing them?

The application also makes it pretty clear that you need to back up settings if you want to be able to access your notes again in the future. This is why support for online backup services is integrated. Given that everything is saved locally in transient locations like localstorage, of course you are going to lose everything if you don't make a more permanent backup.

There are other centralized and encrypted solutions if that's what you're looking for but Laverna is just a nice tool that makes use of convenient web technologies in order to create local encrypted notes. The application itself encourages you use a backup solution during setup so I don't see the issue.

**Comment by Boggin** [comment]
Poor attitude, @Poorchop.

**Comment by smtchahal** [comment]
@dlip I think it's okay to keep the settings file in your Dropbox. Even if someone gets hold of the file, they would still need to enter the correct password in order to access your notes.

Or at least that's how it seems to work, I just tried this a few times.

**Comment by dlip** [comment]
Yeah, I just put it in an encrypted file in dropbox and it works ok. I still think maybe it's too much confusion for the average user though, maybe the salt could be an option that is off by default.

**Comment by mcodexx** [comment]
I have exactly the very same problem as simonkeng. My notes are in my dropbox account but either the web based Laverna and the Laverna app can't read them anymore. That's must be the most stupid note taker I have ever tried. I can see lots of people lost their notes due to this glitch. Has anybody solved the problem with no programming?

**Comment by funilrys** [comment]
Hello there,
Please report to https://github.com/Laverna/laverna/issues/971#issuecomment-411423965 which explain the state of this project.

Have a nice day/night,
Cheers,
Nissar
